Wikipedia/Code/SavedArticlesFetcher.swift

```diff
@@ -196,51 +199,61 @@ private extension SavedArticlesFetcher {
                 articleCacheController.remove(groupKey: articleKey, itemCompletion: { (itemResult) in
                     switch itemResult {
                     case .success(let itemKey):
-                        print("🙈successfully added \(itemKey)")
+                        print("🙈successfully removed \(itemKey)")
                     case .failure(let error):
                         print("🙈failure in itemCompletion of \(articleKey): \(error)")
                     }
                 }) { (groupResult) in
-                    switch groupResult {
-                    case .success:
+                    DispatchQueue.main.async {
+                        switch groupResult {
+                        case .success:
+                            print("🙈success in groupCompletion of \(articleKey)")
+                            self.didRemoveArticle(with: articleKey)
+                            self.updateFetchesInProcessCount()
+                        case .failure:
+                            print("🙈failure in groupCompletion of \(articleKey)")
```

Comment:
@joewalsh we might need a call here to self.updateFetchesInProcessCount()

Reply:
eh on second thought maybe not. If it fails to remove, isDownloaded will remain true thus the in process count won't change.